### PR TITLE
Add find command

### PR DESCRIPTION
**Pull request by floroks**
Adds a find command to the odxtools, allowing you to search &
decode services by hex-byte-data. Also allows searching
by service name.

--

Florian Roks \<florian.roks@daimler.com\>, Daimler TSS GmbH
[Imprint](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)

Licensed under MIT

### CODE REVIEW

File: odxtools/cli/find.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    # Collapse blank lines
+    desc = re.sub(r'(\n\s*)+\n+', '\n', desc).strip()
+
+    if '\n' in desc:
+        desc = '\n' + ident * ' ' + \
+            ('\n' + ident * ' ').join(desc.split('\n'))
+    return desc
+
+
+def print_diagnostic_service(service: DiagService, print_params=False):
```
Comment (andlaus) [inline odxtools/cli/find.py:27]: hm, I guess we should probably move the "print service/parameter/etc" functions to a separate file and use them from all tools which need them (`list` and `browse` come to mind). Anyway, that's for another day. (at least if you don't want to jump on this immediately ;))

**Review by andlaus — COMMENTED** [review]
nice! I only found a few nitpicks...

File: odxtools/cli/find.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2021 MBition GmbH
+
+import argparse
+import re
+import markdownify
```
[the remark this answers is not in the document]
Comment (floroks) [reply odxtools/cli/find.py]: It's already in the requirements.txt (already used in `list` command)

File: odxtools/cli/find.py (hunk 1 above)
Comment (floroks) [reply odxtools/cli/find.py]: I moved the duplicated functions from list and find into _print_utils.py as a start

**Comment by floroks** [comment]
commit was amended & force-pushed to include the requested changes (see conversations)

File: README.md [hunk 3]
```diff
@@ -413,6 +414,68 @@ Tester: do_forward_flips(forward_soberness_check=18, num_flips=50)
...
+    Request byte position = 0, byte length = 1
+   Parameter(short_name='response_code', type='VALUE', semantic=None, byte_position=2, bit_length=8, dop_ref='somersault.DOP.error_code')
+    DataObjectProperty('error_code', category='IDENTICAL', internal_type='A_UINT32', physical_type='A_UINT32')
+
+Decoded Request('start_session'):
+	sid: 16
+	id: 0
+```
+
+`find $BASE_DIR/odxtools/examples/somersault.pdx -d 10 00` would display the same information,
```
Comment (andlaus) [inline README.md:476]: s/find/odxtools find/ (`find` is a standard unix command). Sorry that I missed this in the review...

Comment (floroks) [reply README.md]: No worries, missed it, after copying and pasting the command - fixed and repushed.

**Comment by andlaus** [comment]
all cool now :). Merging, thanks for the contribution!

**Pull request merged**